test/e2e/tests/test_feature_group.py

```diff
@@ -0,0 +1,122 @@
+def feature_group():
+    """Creates a feature group from a SPEC_FILE."""
+    feature_group_name = random_suffix_name(RESOURCE_NAME_BASE, 32)
+    replacements = REPLACEMENT_VALUES.copy()
+    replacements["FEATURE_GROUP_NAME"] = feature_group_name
+    reference, spec, resource = create_sagemaker_resource(
+        resource_plural=RESOURCE_PLURAL,
+        resource_name=feature_group_name,
+        spec_file=SPEC_FILE,
+        replacements=replacements,
+    )
+    assert resource is not None
+    yield (reference, resource)
+    
+    # Delete the k8s resource if not already deleted by tests
+    if k8s.get_resource_exists(reference):
+        _, deleted = k8s.delete_custom_resource(reference, WAIT_PERIOD_COUNT, WAIT_PERIOD_LENGTH)
+        assert deleted
+        
+def get_sagemaker_feature_group(feature_group_name: str):
+    """Used to check if there is an existing feature group with a given feature_group_name."""
+    try:
+        return sagemaker_client().describe_feature_group(FeatureGroupName=feature_group_name)
+    except botocore.exceptions.ClientError as error:
+        logging.error(
+            f"SageMaker could not find a feature group with the name {feature_group_name}. Error {error}"
+        )
+        return None
+    
+def get_feature_group_status(feature_group_name: str):
+    feature_group_describe_response = get_sagemaker_feature_group(feature_group_name)
+    return feature_group_describe_response["FeatureGroupStatus"]
+
+@service_marker
+@pytest.mark.canary
+class TestFeatureGroup:
+    def _wait_feature_group_status(
+            self,
+            feature_group_name,
+            expected_status: str,
+            wait_periods: int = WAIT_PERIOD_COUNT,
+            period_length: int = WAIT_PERIOD_LENGTH,
+    ):
+        return wait_for_status(
+            expected_status,
+            wait_periods,
+            period_length,
+            get_feature_group_status,
+            feature_group_name,
+        )
+    
+    def test_create_feature_group(self, feature_group):
+        """Tests that a feature group can be created and deleted
+        using the Feature Group Controller.
+        """
+        (reference, resource) = feature_group
+        assert k8s.get_resource_exists(reference)
+        
+        feature_group_name = resource["spec"].get("featureGroupName", None)
+        
+        feature_group_describe_response = get_sagemaker_feature_group(feature_group_name)
+        
+        assert (
+            k8s.get_resource_arn(resource)
+            == feature_group_describe_response["FeatureGroupArn"]
+        )
+        
+        assert (
+            self._wait_feature_group_status(
+                feature_group_name, FEATURE_GROUP_STATUS_CREATED
+            )
+        )
+        
+        # Delete the k8s resource.
+        _, deleted = k8s.delete_custom_resource(reference, WAIT_PERIOD_COUNT, WAIT_PERIOD_LENGTH)
+        assert deleted
+        # TODO: Once the delete requeue PR is merged,
```

Comment:
FYI: you are not completely blocked on this
Have you looked at this sample PR? - https://github.com/aws-controllers-k8s/elasticache-controller/pull/34/files

You can implement the logic to requeueOnDelete. only this missing is the resource getting patched after delete is called. This will be enabled once [the PR on runtime is merged](https://github.com/aws-controllers-k8s/runtime/pull/21/files)

Reply:
Added a TODO for implementing the logic to requeueOnDelete.